src/PositionVelocityTime.jl

```diff
@@ -21,30 +23,127 @@ module PositionVelocityTime
         carrier_phase::CP = 0
     end
 
+    """
+    PVTSolution, with used sats and el and az.
+    """
+    @with_kw mutable struct PVTSolution
+        pos::ECEF
+        receiver_time_correction::Float64
+        GDOP::Union{Nothing, Float64}
+        PDOP::Union{Nothing, Float64} 
+        VDOP::Union{Nothing, Float64} 
+        HDOP::Union{Nothing, Float64} 
+        TDOP::Union{Nothing, Float64}
+        #following is added for satllite observation
+        num_used_sats::Int64
+        used_sats::Union{Nothing,AbstractVector{Int64}} 
+        elevation_sats::Union{Nothing,AbstractVector{Float64}}  
+        azimuth_sats::Union{Nothing,AbstractVector{Float64}}  
+    end
+    
+    """
+    Calculates elevation and azimuth of satellite
+
+    $SIGNATURES
+    user_pos_ecef: user position in ecef coordinates
+    sat_pos_ecef: satellite position in ecef coordinates
+
+    This function calculates the elevation and azimuth of the satellite in degree
+    The implementation follows ESA_GNSS-Book_TM-23_Vol_I B.3
+    """
+
+    function get_elevation_azimuth(user_pos_ecef::ECEF, sat_pos_ecef)
+        
+        p = sat_pos_ecef-user_pos_ecef
+        dist = norm(p)
+        p̂ = p/dist
+        
+        user_lla = LLAfromECEF(wgs84)(ECEF(user_pos_ecef[1], user_pos_ecef[2], user_pos_ecef[3]))
+        lat = user_lla.lat
+        lon = user_lla.lon
+
+        ê = [-sind(lat), cosd(lat), 0]
+        n̂ = [-cosd(lon)*sind(lat), -sind(lat)*sind(lon), cosd(lat)]
+        û = [cosd(lon)*cosd(lat), sind(lon)*cosd(lat), sind(lat)]
+
+        el = asind(dot(p̂,û))
+        az = atand(dot(p̂,ê),dot(p̂,n̂))
+
+        return el, az
+
+    end
 
     """
     Calculates ECEF position of user
 
     $SIGNATURES
     ´sat_state´: satellite state, combining decoded data, code- and carrierphase 
+    
+    ´min_elevation´: the minimum elevation for a satellite, that can be used for a positioning
+                    min_elevation = -100, deactivates the  elevation filter
+    
 
     This function calculates the position of the user in ECEF coordinates
+        incl. earth rotation correction
+
+  
     The implementation follows IS-GPS-200K Table 20-IV.
     """
     function calc_PVT(
-        satellite_states::AbstractVector{SatelliteState{Float64}}
+        prev_pvt::PVTSolution,
```

Comment:
In the first iteration, the previous PVT is unknown.
What do you think about, giving this parameter a default value like: `prev_pvt = PVTSolution()`

Reply:
This would also avoid a breaking change in the function signature, as the `prev_pvt` Parameter would have to switch position with `satellite_states`